### PR TITLE
mgr/cephadm: bump monitoring stacks versions

### PR DESCRIPTION
**Pull request by nizamial09**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by adk3798** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
@aaSharma14 @cloudbehl looks like we need to migrate some of the old grafana panels to newer ones. I saw some of them not working with the existing implementation, like the one we have in Pool Overview

Screenshot from 2024-03-21 11-49-04

The `Table (old)` is used many places so we need a separate PR to do that migration

**Comment by nizamial09** [comment]
@ceph/dashboard we'll need to update alertmanager api to start using v2: https://github.com/prometheus/alertmanager/releases/tag/v0.27.0

I re-opened this: https://github.com/ceph/ceph/pull/53679

**Comment by nizamial09** [comment]
@aaSharma14 is taking care of some migrations in grafana so that we can properly start using grafana 10: https://github.com/ceph/ceph/pull/56509

**Comment by nizamial09** [comment]
@adk3798 fyi, the required migration are complete now with the help of @aaSharma14. Once the test started passing, we can proceed to merge this..

File: src/cephadm/cephadmlib/constants.py [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
 DEFAULT_IMAGE_IS_MAIN = True
 DEFAULT_IMAGE_RELEASE = 'squid'
-DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
-DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
-DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
-DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
-DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
-DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/grafana:9.4.12'
+DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.51.0'
+DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.9.5'
```
Comment (ktdreyer) [inline src/cephadm/cephadmlib/constants.py:8]: Guillaume has also been working to update Loki at https://github.com/ceph/ceph/pull/57164

Comment (nizamial09) [reply src/cephadm/cephadmlib/constants.py]: thanks, i removed those from this PR.

**Comment by nizamial09** [comment]
Related e2e failure. Need to adapt the e2e
```
[Test Result](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/10656/testReport/) (2 failures / -2)
[RGW Daemon Overall Performance (example #1).Grafana panels RGW Daemon Overall Performance (example #1)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/10656/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_1)/Grafana_panels_RGW_Daemon_Overall_Performance__example__1_/)
[RGW Daemon Overall Performance (example #6).Grafana panels RGW Daemon Overall Performance (example #6)](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/10656/testReport/junit/(root)/RGW%20Daemon%20Overall%20Performance%20(example%20_6)/Grafana_panels_RGW_Daemon_Overall_Performance__example__6_/)
```

**Review by afreen23 — APPROVED** [review]
for failing tests lets have a tracker, thanks!

**Comment by nizamial09** [comment]
tracker to fix e2e: https://tracker.ceph.com/issues/66491

**Pull request merged**